### PR TITLE
Improving error handing to not ignore or override them

### PR DESCRIPTION
**Pull request by d6o**
## Description
In some situations errors can be override on the loop from UpdateResources function, created a new var to aggregate all error from the function while err var will be used to handle individuals errors from calls

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: go/src/koding/kites/kloud/stack/provider/apply.go [hunk 1]
```diff
@@ -365,12 +365,14 @@ func (bs *BaseStack) UpdateResources(state *terraform.State) error {
 		return err
 	}
 
+	var merr error
```
Comment (rjeczalik) [inline go/src/koding/kites/kloud/stack/provider/apply.go]: You don't need this variable, it already works like that. With the exception of L383 - rename:

```go
if cred, err := bs.Builder.CredentialByProvider(machine.Provider); err == nil
```

to

```go
if cred, e := bs.Builder.CredentialByProvider(machine.Provider); e == nil {
```

Also improve error message to sth like:

```go
err = multierror.Append(err, fmt.Errorf("machine %q: no credential found for %q provider: %s", label, machine.Provider, e))
```

**Comment by DiSiqueira** [comment]
Changes made, i also change in L394 for the same purpose.

**Comment by rjeczalik** [comment]
Can I ask you to also improve the credential error message? L384, see my previous comment.

**Review by rjeczalik — APPROVED** [review]

File: go/src/koding/kites/kloud/stack/provider/apply.go [hunk 2]
```diff
@@ -378,10 +378,10 @@ func (bs *BaseStack) UpdateResources(state *terraform.State) error {
 			continue
 		}
 
-		if cred, err := bs.Builder.CredentialByProvider(machine.Provider); err == nil {
+		if cred, e := bs.Builder.CredentialByProvider(machine.Provider); e == nil {
 			machine.Credential = cred
 		} else {
-			err = multierror.Append(err, fmt.Errorf("machine %q: no credential found for %q provider", label, machine.Provider))
+			err = multierror.Append(e, fmt.Errorf("machine %q: no credential found for %q provider: %s", label, machine.Provider, e))
```
Comment (ppknap) [inline go/src/koding/kites/kloud/stack/provider/apply.go]: I believe you meant: `err = multierror.Append(err, ...)`

File: go/src/koding/kites/kloud/stack/provider/apply.go [hunk 3]
```diff
@@ -391,9 +391,9 @@ func (bs *BaseStack) UpdateResources(state *terraform.State) error {
 			continue
 		}
 
-		err := modelhelper.UpdateMachine(m.ObjectId, bson.M{"$set": bs.buildUpdateObj(machine, state, now)})
-		if err != nil {
-			err = multierror.Append(err, fmt.Errorf("machine %q failed to update: %s", label, err))
+		e := modelhelper.UpdateMachine(m.ObjectId, bson.M{"$set": bs.buildUpdateObj(machine, state, now)})
+		if e != nil {
+			err = multierror.Append(e, fmt.Errorf("machine %q failed to update: %s", label, err))
```
Comment (ppknap) [inline go/src/koding/kites/kloud/stack/provider/apply.go]: It should be:

```
err = multierror.Append(err, fmt.Errorf("machine %q failed to update: %s", label, e))
```

**Review by ppknap — APPROVED** [review]

**Pull request merged**